tests/unit/browsers/test_browser_pool.py

```diff
@@ -160,3 +160,27 @@ async def test_with_plugin_contains_page_options(server_url: URL) -> None:
         await test_page.page.goto(str(server_url / 'user-agent'))
         assert 'My Best User-Agent' in await test_page.page.content()
         await test_page.page.close()
+
+
+@pytest.mark.parametrize(
+    ('page_count', 'equal_browsers'),
```

Comment:
I'd rename the first parameter to `retire_after_page_count` and the second parameter to something like `expect_equal_browsers`. Also maybe you could use `is` and `is not` instead of equality operator.

Also the `id` values in `pytest.param` are kinda misleading.